eng/common/pipelines/templates/steps/sparse-checkout.yml

```diff
@@ -15,15 +15,22 @@ parameters:
 steps:
   - ${{ if not(parameters.SkipCheckoutNone) }}:
       - checkout: none
-
+  - pwsh: |
+      Write-Host "##vso[task.setvariable variable=githubLinkPrefix]https://github.com/"
+    displayName: Set github link without PAT
+    condition: eq(variables['azuresdk-github-pat'], '')
+  - pwsh: |
+      Write-Host "##vso[task.setvariable variable=githubLinkPrefix]https://$(azuresdk-github-pat)@github.com/"
```

Comment:
We need to be very careful with this approach as it may end up leaking our pat in the logs.